### ISSUE TITLE
Species selection: ordering of matches to Common Names

### ISSUE THREAD

**Issue by DavidHepper** [BiologicalRecordsCentre/iRecord#1440]
When selecting a Species, is it possible to list matches in order of priority? When "golden-ringed" is entered, the matches look like this: 
![image](https://user-images.githubusercontent.com/24915271/226593838-f46a7279-32bb-47d2-ab83-caeae53a0409.png)

The temptation is to pick the top entry, which isn't the best option for a genus with only one species ever recorded in the UK.
Demoting matches of any rank other than Species would do the trick in this case.

**Comment by Sam-Amy** [comment]
Another similar request via for this via the [forum](https://irecord.org.uk/forum/feedback-and-suggestions/plants-could-list-taxa-presented-recorders-they-type-be-ordered) "many common taxa get entered only with a genus when recorders are typing in common names. Examples are Primrose, Bramble and Hazel which are clearly correctly identified. When I type Hazel into the system the first item in the list is Hazel (Corylus) and the second Hazel (Corylus avellana). Those not familiar with latin binomials understandably pick the first in the list. In some cases the first on the list is not as informative and does not represent the identification sufficiently....As the list is arranged alphabetically, hybrids can be presented before species in some cases. For instance, if you type in Alka the first return is Alkanet (Anchusa ochroleuca x officinalis = A. x baumaartenii) with Alkanet (Anchusa offinalis) second.  This could be baffling for beginners and the most likely answer should be first."

Strangely though, when I try the example above of Hazel, on the casual record form, I get the full species coming up first. 
![image](https://github.com/BiologicalRecordsCentre/iRecord/assets/104146164/780ff2d7-4bd3-48e3-a4c5-d61a51724690)

@burkmarr do you know what dictates the order names appear and why they might differ at times?

**Comment by DavidHepper** [comment]
Do common names come from UKSI? Could they each have a Drupal 'weight', where the recommended English common name has a weight of zero and anything else a positive value?

**Comment by johnvanbreda** [comment]
Yes, the Common Names come from UKSI. Although we could indeed add a weight, maintaining it would be difficult as there is no similar value in UKSI.

When the search list response is sorted, we use a few custom rules to try and ensure that names which more closely match your search term are listed first. For example if you search "blue" then names that start with blue are first, then names with blue somewhere else in the phrase. 

I've just added code to the query so it now raises the priority of names that are species rank (including ranks just either side of species, e.g. aggregates, s.l. and species hybrid). This works well in that when there is a choice, the species name will appear before other ranks. However in the Golden-ringed and also the Bramble examples that doesn't work as the dragonflies name is incorrectly given rank species - see https://uksi-sandbox.nhm.ac.uk/taxon.php?linkKey=NBNSYS0100012143. Plus Bramble is given the rank species hybrid (which is just below Species in the sorted list of ranks) - https://uksi-sandbox.nhm.ac.uk/taxon.php?linkKey=NHMSYS0000841456. I would expect these names to have the same rank as their accepted names surely?

**Comment by DavidHepper** [comment]
Thanks, John! This is useful. 
I'll report the problem with the rank of 'Golden-ringed dragonflies' to UKSI, as this is indeed their issue, and propose a weight for common names, though that would require a change to your protocol in due course.

**Comment by Sam-Amy** [comment]
Thanks @johnvanbreda , that sounds very useful. And yes I would have thought that in the UKSI a synonym should have the same rank as the recommended name, through in this case it is arguably more helpful that the broader concept is given above, for those people who don't look closely, so I think can wait until the BSBI do a wider update to the UKSI.

Do you know thought why the ordering is not consisten between forms for the same letters typed in? I am still seeing the reverse order for hazel in the 'Enter a list of records' form:
![image](https://github.com/BiologicalRecordsCentre/iRecord/assets/104146164/63b17f94-8126-4909-99f0-3bc5b7f5e791)
